### DISCUSSION TITLE
Riverpod asyncnotifier using for future screens

### DISCUSSION THREAD

**Discussion by eaedev** [rrousselGit/riverpod#2385] [Q&A]
Hi,

Maybe too obvious question but I am not able to set my mind.

I have a asyncnotifier which I know is loaded and used let's say, on the main screen. How can I use that provider which is already loaded in another part off my app? It looks like I have to await for it over and over again. Maybe I do not understand something properly.

I am using code generator.

Thanks in advance!

**Comment by AhmedLSayed9** [comment]
If you're confident that your Async Provider has a value, you can use `ref.watch(provider).requireValue`.

**Reply by eaedev** [reply]
This will still get updates if we do changes on the state?

**Reply by AhmedLSayed9** [reply]
Yes
